src/interaction/sessionrecording.cpp

```diff
@@ -0,0 +1,1207 @@
+    } else {
+        unsigned int numLinesInScript = static_cast<unsigned int>(std::count(scriptToSave.begin(),
+            scriptToSave.end(), '\n'));
+        std::stringstream keyframeLine = std::stringstream();
+        //Add simulation timestamp, timestamp relative, simulation time to recording start
+        keyframeLine << "script ";
+        keyframeLine << sm._timestamp << " ";
+        keyframeLine << (sm._timestamp - _timestampRecordStarted) << " ";
+        keyframeLine << std::fixed << std::setprecision(3) << global::timeManager.time().j2000Seconds();
+        keyframeLine << " ";
+        keyframeLine << (numLinesInScript + 1) << " ";
+        keyframeLine << scriptToSave;
+
+        saveKeyframeToFile(keyframeLine.str());
+    }
+}
+
+void SessionRecording::preSynchronization() {
+    if( _state == sessionState::recording ) {
+        saveCameraKeyframe();
+        if( _usingTimeKeyframes )
+            saveTimeKeyframe();
+    }
+    else if (_state == sessionState::playback) {
+        moveAheadInTime();
+    } else if (_cleanupNeeded) {
+        cleanUpPlayback();
+    }
+}
+
+bool SessionRecording::isRecording() {
+    return (_state == sessionState::recording);
+}
+
+bool SessionRecording::isPlayingBack() {
+    return (_state == sessionState::playback);
+}
+
+bool SessionRecording::playbackAddEntriesToTimeline() {
+    bool parsingErrorsFound = false;
+
+    if (isDataModeBinary()) {
+        unsigned char frameType;
+        bool fileReadOk = true;
+
+        while (fileReadOk) {
+            readFromPlayback(frameType);
+            //Check if have reached EOF
+            if (!_playbackFile) {
+                LINFO(fmt::format("Finished parsing {} entries from playback file {}",
+                    _playbackLineNum - 1, _playbackFilename.c_str()));
+                fileReadOk = false;
+                break;
+            }
+            if (frameType == 'c') {
+                playbackCamera();
+            }
+            else if (frameType == 't') {
+                playbackTimeChange();
+            }
+            else if (frameType == 's') {
+                playbackScript();
+            }
+            else {
+                LERROR(fmt::format("Unknown frame type {} @ index {} of playback file {}",
+                    frameType, _playbackLineNum - 1, _playbackFilename.c_str()));
+                parsingErrorsFound = true;
+                break;
+            }
+
+            _playbackLineNum++;
+        }
+    } else {
+        while (std::getline(_playbackFile, _playbackLineParsing)) {
+            _playbackLineNum++;
+
+            std::istringstream iss(_playbackLineParsing);
+            std::string entryType;
+            if (!(iss >> entryType)) {
+                LERROR(fmt::format("Error reading entry type @ line {} of playback file {}",
+                    _playbackLineNum, _playbackFilename.c_str()));
+                break;
+            }
+
+            if (entryType.compare("camera") == 0) {
```

Comment:
Can use `==` instead for easier readability

Reply:
Changed